File: frontend/src/types/event.ts
export interface Event {
  id: string;
  title: string;
  description: string;
  date: string; // YYYY-MM-DD
  start_time: string; // HH:MM or empty
  end_time: string; // HH:MM or empty
  color: string;
  image_path: string;
  all_day: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateEventPayload {
  title: string;
  description?: string;
  date: string;
  start_time?: string;
  end_time?: string;
  color?: string;
  all_day?: boolean;
  image?: File;
}

File: backend/models/event.go
package models

import (
	"time"
)

type Event struct {
	ID          string    `json:"id" gorm:"primaryKey"`
	Title       string    `json:"title" gorm:"not null"`
	Description string    `json:"description"`
	Date        string    `json:"date" gorm:"not null"` // format: YYYY-MM-DD
	StartTime   string    `json:"start_time"`           // format: HH:MM, empty = all day
	EndTime     string    `json:"end_time"`             // format: HH:MM
	Color       string    `json:"color" gorm:"default:'#3B82F6'"`
	ImagePath   string    `json:"image_path"`
	AllDay      bool      `json:"all_day" gorm:"default:true"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
